### PR TITLE
minimal scheduler

### LINKED ISSUE #1: Scheduler: Simple
We need to turn on the lights and the pump each on their own schedule.

The schedule will be set with a YAML (or maybe a TOML or JSON) file.

The simplest schedule would be:
-lights on at time T0 and off at T1 each day.
-pump runs every X minutes out of every Y minutes.

This is planned to be two async tasks.

Signalling to hardware will be an HTTP request.

### PR DESCRIPTION
**Pull request by rustflower**
Work In Progress.

aims to fix #1

### CODE REVIEW

**Comment by rustflower** [comment]
I'll be looking at feedback provided on #1 and #2 and I'll make a minimal scheduler following the provided ideas there.

**Comment by rustflower** [comment]
I think this is functioning, following the suggestions, and is ready for a review.

**Comment by rustflower** [comment]
One possible issue that I noticed. While sorting isn't a problem because each event has its own spawned thread, there is no rust logic stopping multiple, different events for the same device to be set for identical times. 

So, minimally I think this should be in documentation about passing "good" JSON into this crate, but maybe there should also be some rust that warns a user if it detects such simultaneously scheduled events.

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            .expect("reading in JSON to work")),
+    )
+    .expect("JSON resolves to Vec<Device>");
+    println!("devices as Vec<Device> : {:?}", devices);
+
+    let mut set: JoinSet<()> = JoinSet::new();
+
+    for device in devices {
+        println!("working with device {:?}", device.name);
+        let evs = device.events.clone();
```
Comment (SebKuzminsky) [inline src/main.rs:36]: Nit: probably don't need to clone here, just iterate over the original events vec.

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        println!("events : {:?}", evs);
+
+        for e in evs {
+            println!("about to spawn 'e' on inner_set {:?}", e);
+            set.spawn(event_loop(device.clone(), e));
```
Comment (SebKuzminsky) [inline src/main.rs:41]: Nit: probably don't need to clone the device here, pass in a shared ref instead.

Comment (SebKuzminsky) [inline src/main.rs:41]: It's nice and ergonomic that `spawn()` is a method on the `JoinSet`.

**Comment by SebKuzminsky** [comment]
Couple of nitpicky comments but this looks great to me.

**Comment by SebKuzminsky** [comment]
This passes my eyeball test and a quick run test.

**Pull request merged**